Assets/BowAndArrowImproved/Scripts/Enemies/BAAIEnemy.cs

```diff
@@ -29,15 +29,20 @@ private void OnEnable()
         PlayWarCry();
     }
 
-    public void Die(bool playSound)
+    public void Die(bool deathByArrow)
     {
         gameObject.SetActive(false);
 
-        if (playSound)
+        if (deathByArrow)
         {
-            AudioSource.PlayClipAtPoint(breakClips[Random.Range(0, breakClips.Count)], transform.position);    
+            AudioSource.PlayClipAtPoint(breakClips[Random.Range(0, breakClips.Count)], transform.position);
+            UpdateScoreOnDeath scriptComponent = gameObject.GetComponent<UpdateScoreOnDeath>();
+            if (scriptComponent != null)
+            {
+                onDeath.AddListener( scriptComponent.UpdateScoreboard);
```

Comment:
you are adding a listener for `onDeath` right before calling `Die()`? wtf is this? what are you trying to do? why not listen to the event like a normal person when the objects is started? very weird...

Reply:
Removed it and use a boolean in script instead to check if update of score is valid - it will only be valid if the enemy got hit by attack